src/nvim/os/fs.c

```diff
@@ -461,7 +461,7 @@ ptrdiff_t os_read(const int fd, bool *ret_eof, char *const ret_buf,
   while (read_bytes != size) {
     assert(size >= read_bytes);
     const ptrdiff_t cur_read_bytes = read(fd, ret_buf + read_bytes,
-                                          size - read_bytes);
+                                          (unsigned int)(size - read_bytes));
```

Comment:
Do not use `int` here, if you need `unsigned` write that as such. Though on linux `read()` accepts `size_t` in this position and that should be exactly what is being supplied.

Reply:
Done.